src/typechecker/environment.lisp

```diff
@@ -47,93 +62,106 @@
           ('coalton:Boolean
            (type-entry
             :name 'Boolean
-            :runtime-type 'boolean
+            :runtime-type 'cl:boolean
             :type tBoolean
-            :compressed-type t))
+            :compressed-type t
+            :newtype nil))
 
           ('coalton:Char
            (type-entry
             :name 'Char
             :runtime-type 'character
             :type tChar
-            :compressed-type nil))
+            :compressed-type nil
+            :newtype nil))
 
           ('coalton:I32
            (type-entry
             :name 'I32
             :runtime-type '(signed-byte 32)
             :type tI32
-            :compressed-type nil))
+            :compressed-type nil
+            :newtype nil))
 
           ('coalton:I64
            (type-entry
             :name 'I64
             :runtime-type '(signed-byte 64)
             :type tI64
-            :compressed-type nil))
+            :compressed-type nil
+            :newtype nil))
 
           ('coalton:U8
            (type-entry
             :name 'U8
             :runtime-type '(unsigned-byte 8)
             :type tU8
-            :compressed-type nil))
+            :compressed-type nil
+            :newtype nil))
 
           ('coalton:U32
            (type-entry
             :name 'U32
             :runtime-type '(unsigned-byte 32)
             :type tU32
-            :compressed-type nil))
+            :compressed-type nil
+            :newtype nil))
 
           ('coalton:U64
            (type-entry
             :name 'U64
             :runtime-type '(unsigned-byte 64)
             :type tU64
-            :compressed-type nil))
+            :compressed-type nil
+            :newtype nil))
 
           ('coalton:Integer
            (type-entry
             :name 'Integer
```

Comment:
These type names should be corrected.

```
:name 'coalton:Integer￼
:type 'cl:integer
```

Likewise for the rest (character, *-float, string, etc)